content/swiftui/concepts/drag-and-drop/drag-and-drop.md

```diff
@@ -0,0 +1,155 @@
+    // Triggered when an object enters the view.
+  } 
+    func dropExited(info: DropInfo) {
+    // Triggered when an object exits the view.
+  }
+
+    func dropUpdated(info: DropInfo) -> DropProposal? {
+    // Triggered when an object moves within the view. 
+  }
+  
+    func validateDrop(info: DropInfo) -> Bool {
+    // Determines whether to accept or reject the drop.
+  }
+
+    func performDrop(info: DropInfo) -> Bool {
+    // Handles the drop when the user drops an object onto the view.
+  }
+}
+```
+## DropProposal
+
+**`DropProposal`** is a `structure` in SwiftUI that helps to control how object and drop operations should be handled when it's dropped into a SwiftUI view. It is used together with the onDrop() modifier to determine what happens when a user drops an object.
+You can customize the DropProposal using the operation property to specify the drop operation (e.g., .move or .copy).
+
+```swift
+struct ContentView: View {
+    @State private var text: String = ""
+    
+    var body: some View {
+        VStack {
+            Text("DropProposal Example Codecademy")
+                .font(.largeTitle)
+                .padding()
+            
+            Text("Drop Text Here:")
+            
+            Text(text)
+                .padding()
+                .background(Color.yellow)
+                .onDrop(of: [UTType.text], isTargeted: $text?, perform: { providers, isTargeted in
+                        // Create a custom DropProposal
+                        let dropProposal = DropProposal(operation: .copy)
+                        
+                        // Check if the drop is targeted to this view
+                        if isTargeted {
+                            // Handle the drop operation based on the custom DropProposal
+                            if let itemProvider = providers.first {
+                                itemProvider.loadObject(ofClass: String.self) { item, error in
+                                    if let droppedText = item as? String {
+                                        // Update the text with the dropped content
+                                        DispatchQueue.main.async {
+                                            self.text = droppedText
+                                        }
+                                    }
+                                }
+                                return dropProposal
+                            }
+                        }
+                        
+                        return nil
+                })
+                .frame(width: 200, height: 100)
+                .border(Color.gray)
+        }
+    }
+}
+
+```
+
+## DropInfo
+
+**`DropInfo`** is a `structure` that gives information about a drop action. It is commonly used when handling drops with the onDrop() modifier. DropInfo gives details about the drop location, the types of data being dropped, and other important information.
+
+## Drag and Drop in SwiftUI example
+
```

Comment:
Please add a sentence or two introducing the example and summarizing what it is doing.